mainapp/views.py

```diff
@@ -1,32 +1,11 @@
 from django.shortcuts import render
-from django.views.generic import TemplateView
-from django.http import HttpResponseRedirect
-from django.contrib.auth import authenticate, login
-from django.urls import reverse
-
-
-class LoginView(TemplateView):
-    template_name = "login.html"
-
-    def post(self, request, *args, **kwargs):
-        username = request.POST["username"]
-        password = request.POST["password"]
-        user = authenticate(request, username=username, password=password)
-
-        if user is not None:
-            login(request, user)
-            return HttpResponseRedirect(reverse("index"))
-
-        else:
-            return HttpResponseRedirect(reverse("login"))
+from django.contrib.auth.decorators import login_required
 
 
 def index(request):
-    return render(request, "index.html")
+    return render(request, "mainapp/index.html")
 
 
+@login_required
```

Comment:
Cool!

Reply:
yeah I thought that was super nifty also.